pkg/controller/controlplane/valuesprovider.go

```diff
@@ -499,37 +555,38 @@ func getCalicoBgpChartValues(
 	var serviceLbIPs, serviceExtIPs, serviceClusterIPs []string
 	var peers []map[string]any
 	var filters []map[string]any
-	if cpConfig.LoadBalancerConfig.CalicoBgpConfig != nil &&
+	var calicoBgpConfig = cpConfig.LoadBalancerConfig.CalicoConfig.CalicoBgpConfig
+	if cpConfig.LoadBalancerConfig.CalicoConfig == nil || cpConfig.LoadBalancerConfig.CalicoConfig.CalicoBgpConfig != nil &&
```

Comment:
The assignment should happen after ensuring `cpConfig.LoadBalancerConfig.CalicoConfig` is not nil.